pkg/azurefile/controllerserver.go

```diff
@@ -825,68 +826,130 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		mc.ObserveOperationWithResult(isOperationSucceeded, SourceResourceID, sourceVolumeID, SnapshotName, snapshotName)
 	}()
 
-	exists, item, err := d.snapshotExists(ctx, sourceVolumeID, snapshotName, req.GetSecrets())
-	if err != nil {
+	if len(req.GetSecrets()) > 0 {
+		exists, item, err := d.snapshotExists(ctx, sourceVolumeID, snapshotName, req.GetSecrets())
```

Comment:
e.g. you could extend `snapshotExists(ctx, sourceVolumeID, snapshotName, req.GetSecrets())` func, make it work for both management api and dataplane api, and check whether `len(req.GetSecrets()) > 0` inside `snapshotExists`